fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -451,20 +452,27 @@ public static SearchParameter getSearchParameter(String resourceType, String cod
             Set<SearchParameter> params = getSearchParametersByCodeFromTenantOrBuiltIn(resourceType, code, tenantSpMap);
 
             if (params != null && !params.isEmpty()) {
-                result = params.iterator().next();
-                if (params.size() > 1) {
-                    log.warning("Found multiple resource-specific search parameters for code '" + code + "' on resource type " + resourceType + ";"
-                            + " use search parameter filtering to disambiguate. Using '" + result.getUrl().getValue() + "'.");
+                Iterator<SearchParameter> iterator = params.iterator();
+                result = iterator.next();
+                String configuredUrl = result.getUrl().getValue();
+                while (iterator.hasNext()) {
+                    SearchParameter conflict = iterator.next();
+                    log.warning("Found multiple resource-specific search parameters, '" + configuredUrl + "' and '" + conflict.getUrl().getValue()
+                        + "', for code '" + code + "' on resource type '" + resourceType + "';"
+                        + " use search parameter filtering to disambiguate. Using '" + configuredUrl + "'.");
                 }
             }
         } else if (parentResourceFilterRules == null || parentResourceFilterRules.containsKey(SearchConstants.WILDCARD)) {
             Set<SearchParameter> params = getSearchParametersByCodeFromTenantOrBuiltIn(SearchConstants.RESOURCE_RESOURCE, code, tenantSpMap);
 
             if (params != null && !params.isEmpty()) {
-                result = params.iterator().next();
-                if (params.size() > 1) {
-                    log.warning("Found multiple cross-resource search parameters for code '" + code + "';"
-                            + " use search parameter filtering to disambiguate. Using '" + result.getUrl().getValue() + "'.");
+                Iterator<SearchParameter> iterator = params.iterator();
+                result = iterator.next();
+                String configuredUrl = result.getUrl().getValue();
+                while (iterator.hasNext()) {
+                    SearchParameter conflict = iterator.next();
+                    log.warning("Found multiple cross-resource search parameters, '" + configuredUrl + "' and '" + conflict.getUrl().getValue()
+                        + "', for code '" + code + "'; use search parameter filtering to disambiguate. Using '" + configuredUrl + "'.");
```

Comment:
it feels like this message should have an option to silence it.  It could be rather chatty if one has a decent sized Search Bundle.

Reply:
The original idea behind the warning was that operators *should* use our fhir-server-config support to indicate which search parameter they want us to use in case of conflicts.  So the way to silence it is to "use search parameter filtering to disambiguate".

What this task is doing is putting enough information in the message so that the config change is as easy as possible.

Aside:  because search parameter filtering is done on a per-tenant basis, we can't easily move these warnings to startup unless we start looping through the various tenant configurations during startup.  I had put some related thoughts for that into https://github.com/IBM/FHIR/issues/1625

Reply:
i don't think we have more than 2 conflicting params  too often, but maybe it would help to collect the various conflicts into a single message instead of logging a different warning for each one?

for example:
```
log.warning("Found multiple cross-resource search parameters for code '" + code + "';"
                            + " use search parameter filtering to disambiguate. Using '" + result.getUrl().getValue() + "'. "
                            + result.stream().map(sp -> sp.getUrl().getValue()).collect(Collectors.joining(", ") );
```

Reply:
@prb112 Are you ok with this as is?